File: HackerEarth/Arrays/MonkAndRotation.java
import java.util.*;

class MonkAndRotation {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of test cases - ");
        int t = sc.nextInt();
        System.out.println("Enter the number of elements 'n' and enter those elements - ");
        for(int i=0; i<t; i++){
            int n = sc.nextInt();
            int[] arr = new int[n];
            for(int j=0; j<n; j++){
                arr[j] = sc.nextInt();
            }
            System.out.print("Enter the value of k - ");
            int k = sc.nextInt();
            System.out.println("Rotated array is - " + Arrays.toString(helper(arr,k)));
        }
        sc.close();
    }

    static int[] helper(int[] arr, int k){
        if(arr.length == 0) return new int[]{};
        if(arr.length == 1) return arr;

        int[] res = new int[arr.length];
        int idx = 0;
        int l = arr.length-1;
        k = k%(l+1);
        for(int i=1; i<arr.length+1; i++){
            res[idx++] = arr[(l-k+i)%(l+1)];
        }

        return res;
    }
}

File: HackerEarth/Arrays/MonkandRotation.py
"""
Monk loves to preform different operations on arrays, and so being the principal of Hackerearth School, he assigned a task to his new student Mishki. Mishki will be provided with an integer array A of size N and an integer K , where she needs to rotate the array in the right direction by K steps and then print the resultant array. As she is new to the school, please help her to complete the task.

Video approach to solve this question: here

Input:
The first line will consists of one integer T denoting the number of test cases.
For each test case:
1) The first line consists of two integers N and K, N being the number of elements in the array and K denotes the number of steps of rotation.
2) The next line consists of N space separated integers , denoting the elements of the array A.

Output:
Print the required array.
"""
import typing

class Solution:
    def helper(self, nums, k) -> list:
        n = len(nums)
        k = k % n
        return nums[-k:] + nums[:-k]

t = int(input())
solution = Solution()

for _ in range(t):
    n, k = map(int, input().split())
    nums = list(map(int, input().split()))
    
    result = solution.helper(nums, k)
    print(*result)